backend/test/Test/Setup.hs

```diff
@@ -49,4 +49,7 @@ withContext = around withContext'
             ecDb . dbConnString .~ connString
       withPostgresConn testConfig $ \connPool -> do
         let ctx = EdnaContext testConfig connPool
-        callback ctx <* resetDbSchema ctx
+        callback ctx
+
+ednaTestMode :: EdnaContext -> PropertyT Edna a -> PropertyT IO ()
+ednaTestMode ctx = void . hoist (\action -> runRIO ctx $ schemaInit *> action <* runPg resetSchema)
```

Comment:
Also what do you think about my suggestion to wrap it into `bracket` or do `resetSchema` before `schemaInit`?

Also for SMT we also need to do `schemaInit` and `resetSchema`, but there we can't use `ednaTestMode`, so it would be nice to extract `schemaInit *> action <* runPg resetSchema` into another function.

Reply:
What is the difference between `schemaInit` before `resetSchema` and `resetSchema` before `schemaInit`? And how `bracket` solve this issue?

Reply:
Currently if `action` throws an uncaught exception, `resetSchema` won't run. But I suppose this exception may be caught by testing engine afterwards (I am not sure if all exceptions are caught, but at least things like `HUnitFailure` should be caught for sure) and then the next test will run, but schema won't be reset when we run it.

If we use `bracket`, `resetSchema` will always happen. Or we can move it before `schemaInit` and then `resetSchema` will always happen if `schemaInit` happens.

Reply:
I made it in a little bit different way